server/graphql-schema.js

```diff
@@ -65,40 +65,108 @@ const graphqlSchema = gql`
     }
 
     type TestPlan {
+        """
+        Corresponds to the ARIA-AT directory storing the test plan.
+        """
         id: ID!
+        atGitSha(gitSha: String!): TestPlanVersion
+        latestVersion: TestPlanVersion!
+        latestVersionWithResults: TestPlanVersion
+        versions: [TestPlanVersion]!
+        versionsWithResults: [TestPlanVersion]!
+        queueRelevantVersions: [TestPlanVersion]!
+    }
+
+    type TestPlanVersion {
         title: String!
-        publishStatus: TestPlanStatus!
-        revision: String!
-        sourceGitCommit: String!
+        status: TestPlanStatus!
+        gitSha: String!
+        gitMessage: String!
         exampleUrl: String!
-        # TODO: consider renaming createdAt and including the date the source
-        # code age was authored as well.
-        createdAt: Timestamp!
+        updatedAt: Timestamp!
         tests: [Test]!
         testCount: Int!
+        testPlanReports: [TestPlanReport]!
+        testPlanReport(id: ID, testPlanTarget: ID): TestPlanReport
     }
 
     interface Test {
         title: String!
-        totalAssertions: Int!
-        # TODO: need complete representation of test data
+        index: Int!
+        # TODO: account for running scripts
+        instructions: [Instruction]!
+        assertions: [Assertion]!
+        assertionCount: Int!
+        optionalAssertionCount: Int!
+        passThroughs: [PassThrough]!
+    }
+
+    type Instruction {
+        # TODO: account for automation
+        manualInstruction: String!
+    }
+
+    interface Assertion {
+        # TODO: account for at-specific assertions
+        # TODO: account for optional assertions
+        # TODO: account for automation
+        manualAssertion: String!
+    }
+
+    interface PassThrough {
+        atMode: String!
+        nthCommand: Int!
```

Comment:
The test authors have a way to specify a group of commands which should be interchangeable within a certain test. In practice it seems that pretty much every interaction we're testing can be accomplished in at least a few different ways.

So in order to completely run a test, manual testers have to re-run it once for each available and interchangeable command. This could mean three, four, eight or more "pass throughs."

nthCommand, then, is used to cycle through all the available commands, and it would apply to all the groups of commands which are used within the test.

It's possible the "nthCommand" part might get replaced with an actual command type in the future... it is still a tiny bit blurry!

Reply:
Got it! Thanks for the explanation.